### PR TITLE
Merge UserInfoService and UserService

### PR DESCRIPTION
**Pull request by xcoulon**
also refactored the controllers to use application.Application
to load the services.

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

**Comment by fabric8cd** [comment]
@xcoulon snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-593-1`

File: account/service/tenant.go [hunk 1]
```diff
@@ -16,28 +16,28 @@ import (
...
 )
 
-type tenantConfig interface {
-	GetTenantServiceURL() string
-}
-
-// Tenant represents Tenant Service
-type Tenant interface {
+// TenantService represents the Tenant service
+type TenantService interface {
```
Comment (sbose78) [inline account/service/tenant.go]: Good cleanup of old inconsistent naming.

File: account/service/user.go [hunk 2]
```diff
@@ -2,21 +2,65 @@ package service
...
+	tokenManager token.Manager
+}
+
+// UserInfo gets user information given a context containing access_token
+func (s *userServiceImpl) UserInfo(ctx context.Context, identityID uuid.UUID) (*account.User, *account.Identity, error) {
+	var user *account.User
+	var identity *account.Identity
+	err := s.ExecuteInTransaction(func() error {
+		var err error
+		identity, err = s.Repositories().Identities().Load(ctx, identityID)
```
Comment (sbose78) [inline account/service/user.go]: We have a method called IdentityWithUser(..) . Could we use that here ?

File: application/service/services.go [hunk 3]
```diff
@@ -75,6 +75,7 @@ type SpaceService interface {
 
 type UserService interface {
 	DeprovisionUser(ctx context.Context, username string) (*account.Identity, error)
+	UserInfo(ctx context.Context, identityID uuid.UUID) (*account.User, *account.Identity, error)
```
Comment (sbose78) [inline application/service/services.go]: Nice! looks clean now :)

**Comment by sbose78** [comment]
.. and you have a conflict with your own PR I merged , @xcoulon :fireworks:

File: account/service/user.go (hunk 2 above)
Comment (xcoulon) [reply account/service/user.go]: Let me see, indeed

**Comment by xcoulon** [comment]
thanks for the review, @sbose78. Let me take care of the merge conflict

File: account/service/tenant.go (hunk 1 above)
Comment (xcoulon) [reply account/service/tenant.go]: thanks ;)

File: application/service/services.go (hunk 3 above)
Comment (xcoulon) [reply application/service/services.go]: thanks ;)

**Comment by fabric8cd** [comment]
@xcoulon snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-593-2`

File: account/service/user.go (hunk 2 above)
Comment (xcoulon) [reply account/service/user.go]: @sbose78 I applied your suggestion in https://github.com/fabric8-services/fabric8-auth/pull/593/commits/7b7fda4942d86110c04f46c53645acb42246b163

**Comment by fabric8cd** [comment]
@xcoulon snapshot fabric8-auth image is available for testing.  `docker pull docker.io/fabric8/fabric8-auth:SNAPSHOT-PR-593-3`

**Pull request merged**